design_docs/support_external_swift_packages.md

```diff
@@ -0,0 +1,204 @@
+# Proposal for Representing External Swift Packages
+
+## Problem
+
+External Swift packages are, typically, represented in Bazel projects in one of two ways:
+
+1. As an external repositry using `http_archive` providing a custom build file.
+2. As an external repository using `spm_repositories` and `spm_pkg` from
+   [cgrindel/rules_spm](https://github.com/cgrindel/rules_spm/). 
+
+If an Xcode project is generated to build with Bazel (BWB), no additional support is required in the
+Xcode project for the project to build properly. Whatever is defined in the Bazel project will
+handle the building and linking of the external dependencies.
+
+If an Xcode project is generated to build with Xcode (BWX), the Xcode project needs to understand
+how to incorporate the external dependency.
+
+## Possible Solutions
+
+### Solution 1: Treat the External Dependency as Another Target
+
+In essence, the external dependencies are imported as Xcode targets. 
+
+The pros for this approach are:
+
+- The generation code can leverage much of its existing logic mapping Bazel targets.
+- This maps well for Bazel projects that download the external package using `http_archive` and
+  define a custom build file.
+
+The cons for this approach are:
+
+- The generated project will be cluttered with targets that are not relevant to the developer.
+  These will include not just the direct dependencies, but all of the transitive dependencies for
+  the external packages.
+- The generation code will need to handle the source for these external dependencies, differently.
+  It will need to download the source for the external dependencies, map it and make it available in
+  the user interface.
```

Comment:
How is this different from today?

Reply:
The xcodeproj rules currently create targets for source files that exist in the target project (i.e., where you are running the generator code). The source code for the external dependencies does not live in the target project. So, the generator code will need to retrieve the source probably via `bazel fetch` and map the files. Perhaps, this will not be a heavy lift. However, I am sure that there will be some nuances. Also, I am not sure that most developers will want to have it included in the project's navigator pane.

Reply:
I just noticed that we do have an example that uses an external lib (`//examples/cc/tool`). So, perhaps, there aren't as many nuances as I had anticipated.